File: HackerRank-10-Days-Statistics/Day 5/Day_5_Poisson_Distribution_I.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
import math
import sys
a,b = sys.stdin.readlines()
lamda = float(a)
k=int(b)

def factorial(n):
    if (n==0 or n==1):
        return 1
    return n*factorial(n-1)

print(round((lamda**k * math.exp(-lamda))/factorial(k),3))

File: HackerRank-10-Days-Statistics/Day 5/Day_5_Poisson_Distribution_I.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include<bits/stdc++.h>
using namespace std;


int factorial(int n){
    if(n==0 || n==1)
        return 1;
    return n*factorial(n-1);
}


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    float lambda;
    int k;
    cin>>lambda>>k;
    
    double p=pow(lambda,k)*exp(-lambda)/factorial(k);
    cout<<fixed<<setprecision(3)<<p<<endl;  
    return 0;
}
